src/lib/Embed.hs

```diff
@@ -715,6 +741,25 @@ indexToIntE ty idx = case ty of
     imul rSize lIdx >>= iadd rIdx
   TC (IntRange _ _)     -> indexAsInt idx
   TC (IndexRange _ _ _) -> indexAsInt idx
+  RecordTy types -> do
+    sizes <- traverse indexSetSizeE types
+    (strides, _) <- traverseFunM
+      (\sz prev -> do {v <- imul sz prev; return (prev, v)}) sizes (IntVal 1)
```

Comment:
Hmm one interesting question to ask would be whether we want to support dependent record types, and if the answer is yes, then this code will need to be re-worked a little.

Reply:
Dependent in the sense that one record field's type depends on the value from another field? I wonder if that would be useful. I'm not sure how you'd even write the type for that, since at least right now the labels of the record don't actually bind the label names to anything (although they kind of look like they do, I guess, which maybe is confusing), and there's no syntactical ordering imposed on the labels (so we can't say that earlier things are "in scope" for later things, because that depends on how the user wrote the type down).

My gut feeling is that if you want dependent things you should define an ADT, at least for now.

Reply:
I agree that we should only do dependent constructors via ADTs for now. For a general dependent pair, you need some sort of annotation to indicate what the type of the second element is (e.g. `(2, [1,2])` could have type `(Int & (Fin 2 => Int))` or `(n:Int & (Fin n => Int))`. Extensible records are tricky enough as they are!